tools/serverpod_cli/lib/src/config/config.dart

```diff
@@ -34,7 +34,21 @@ class GeneratorConfig {
     required List<String> relativeDartClientPackagePathParts,
     required this.modules,
     required this.extraClasses,
-  }) : _relativeDartClientPackagePathParts = relativeDartClientPackagePathParts;
+    required this.serializeEnumValuesAsStrings,
+  }) : _relativeDartClientPackagePathParts =
+            relativeDartClientPackagePathParts {
+    assert(_instance == null, 'GeneratorConfig should only be loaded once.');
+    _instance = this;
+  }
+
+  /// The current [GeneratorConfig] instance.
+  static GeneratorConfig? _instance;
+
+  /// Get the current [GeneratorConfig] instance.
+  static GeneratorConfig get instance {
+    assert(_instance != null, 'GeneratorConfig is not loaded.');
+    return _instance!;
+  }
```

Comment:
Note: I'm not sure about the singleton approach for the generator config. My hunch tells me it should rather live in one place than be confined as a singleton.

But this might be the right approach since it is accessed on different abstraction levels in multiple places in the code right now.

Reply:
Totally agree, I hate this part of the PR. The alternative was to pass the config value all over the source tree. I tried doing this and ended up modifying dozens and dozens of files to get the value where I needed it to be. It ended up atrocious.